### PR TITLE
Add store to CMP

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/1125189844152671/1202705137400821

### Description
This PR adds a store module to handle a couple of basic autoconsent settings and integrates them in the basic messages that we already had.
### Steps to test this PR
- [x] Install the app and go to SERP/
- [x] In the dax dialog click Hide an. Hide tips forever. 

_First popup handled_
- [x] Navigate to `http://privacy-test-pages.glitch.me/features/autoconsent/`.
- [x] You should see a new Dax Dialog and in the website a button that says `Reject all`
- [x] Click on `Manage Cookie Pop-ups`.
- [x] Dax dialog should disappear and the button should say `I was clicked`.
- [x] Refresh the page.
- [x] Button should say `I was clicked` but no Dax dialog should be shown again.
- [x] Fresh install the app
- [x] Navigate to `http://privacy-test-pages.glitch.me/features/autoconsent/`.
- [x] You should see a new Dax Dialog and in the website a button that says `Reject all`
- [x] Click on `No Thanks`.
- [x] Dax dialog should disappear and the button should say `Reject all`.
- [x] Refresh the page.
- [x] Button should say `Reject all` but no Dax dialog should be shown again.

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2128** <a href="https://app.graphite.dev/github/pr/duckduckgo/android/2128" target="_blank">Graphite</a> 
    * **PR #2129** <a href="https://app.graphite.dev/github/pr/duckduckgo/android/2129" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/android/2129?utm_source=stack-comment).

File: autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/di/AutoconsentModule.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import dagger.SingleInstanceIn
+
+@ContributesTo(AppScope::class)
+@Module
+object AutoconsentModule {
+    @SingleInstanceIn(AppScope::class)
```
Comment (aitorvs) [inline autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/di/AutoconsentModule.kt:30]: Maybe doesn't need to be `Singleton`?

File: autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/RealAutoconsentTest.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+class RealAutoconsentTest {
+
+    private val pluginPoint = FakePluginPoint()
+    private val repository = FakeRepository()
+    private val mockWebView: WebView = mock()
+
+    lateinit var testee: RealAutoconsent
```
Comment (aitorvs) [inline autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/RealAutoconsentTest.kt:36]: nit: never told anybody but I hate the `testee` name. It forces me to go and check what's the type. I generally prefer to name variables after the type, ie. `realAutoconsent` or `autoconsent` in this case

File: autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/handlers/InitMessageHandlerPluginTest.kt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class InitMessageHandlerPluginTest {
+    @get:Rule var coroutineRule = CoroutineTestRule()
+
+    private val mockCallback: AutoconsentCallback = mock()
+    private val webView: WebView = WebView(InstrumentationRegistry.getInstrumentation().targetContext)
+    private val repository = FakeRepository()
+
+    private val testee = InitMessageHandlerPlugin(coroutineRule.testScope, coroutineRule.testDispatcherProvider, repository)
```
Comment (aitorvs) [inline autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/handlers/InitMessageHandlerPluginTest.kt:46]: nit: two files later, I continue hating the `testee` naming convention :D

File: autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/handlers/PopUpFoundMessageHandlerPluginTest.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class PopUpFoundMessageHandlerPluginTest {
+    @get:Rule var coroutineRule = CoroutineTestRule()
+
+    private val mockCallback: AutoconsentCallback = mock()
+    private val webView: WebView = WebView(InstrumentationRegistry.getInstrumentation().targetContext)
+    private val repository = FakeRepository()
+
+    private val testee = PopUpFoundMessageHandlerPlugin(coroutineRule.testScope, coroutineRule.testDispatcherProvider, repository)
```
Comment (aitorvs) [inline autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/handlers/PopUpFoundMessageHandlerPluginTest.kt:44]: nit: yup, still hating it

File: autoconsent/autoconsent-store/src/main/java/com/duckduckgo/autoconsent/store/AutoconsentSettingsDataStore.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        set(value) {
+            preferences.edit(commit = true) {
+                putBoolean(AUTOCONSENT_FIRST_POPUP_HANDLED, value)
+            }
+        }
+
+    companion object {
+        const val FILENAME = "com.duckduckgo.autoconsent.store.settings"
+        const val AUTOCONSENT_USER_SETTING = "AutoconsentUserSetting"
+        const val AUTOCONSENT_FIRST_POPUP_HANDLED = "AutoconsentFirstPopupHandled"
```
Comment (aitorvs) [inline autoconsent/autoconsent-store/src/main/java/com/duckduckgo/autoconsent/store/AutoconsentSettingsDataStore.kt:53]: nit: haven't checked but, can these be `private`?

File: autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/di/AutoconsentModule.kt (hunk 1 above)
Comment (marcosholgado) [reply autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/di/AutoconsentModule.kt]: 👍

File: autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/RealAutoconsentTest.kt (hunk 2 above)
Comment (marcosholgado) [reply autoconsent/autoconsent-impl/src/test/java/com/duckduckgo/autoconsent/impl/RealAutoconsentTest.kt]: Yeah I can agree with that :D

**Review by aitorvs — APPROVED** [review]
LGTM! tested as per description

**Pull request merged**